tick/ast/node.go

```diff
@@ -981,6 +1024,101 @@ func newProgram(p position) *ProgramNode {
 	}
 }
 
+func NewProgramNodeFromTickscript(tickscript string) (*ProgramNode, error) {
```

Comment:
FWIW The reasons I don't like the name is without reading the function's source I thought it was doing some kind of special operation to return a ProgramNode from a TICKscript instead of just using the Parse method.

Another option would be to move this method to the package where it is consumed as a helper method.